lib/pbench/cli/server/options.py

```diff
@@ -22,13 +22,15 @@ def _pbench_server_config(f: Callable) -> Callable:
 
     def callback(ctx, param, value):
         clictx = ctx.ensure_object(CliContext)
-        clictx.config = value
+        clictx.config = (
+            value if value else "/opt/pbench-server/lib/config/pbench-server.cfg"
+        )
         return value
 
     return click.option(
         "-C",
         "--config",
-        required=True,
+        required=False,
         envvar="_PBENCH_SERVER_CONFIG",
         type=click.Path(exists=True, readable=True),
         callback=callback,
```

Comment:
Click is weird. It actually didn't seem to make much difference; but since we *don't* want to require `-C`/`--config` even in the old case, it didn't make much sense.

Reply:
Before I read the results of your exploration above, I was going to suggest that, while the _option_ isn't required, it's _value_ **is**, but the callback provides it if the other sources don't...but, from what you said above, if we stick with `required=True` then we don't get all the way to the callback in certain cases, so I guess `False` is the better choice.